language/vm/vm_runtime/src/data_cache.rs

```diff
@@ -265,8 +283,21 @@ impl<'txn> TransactionDataCache<'txn> {
         }
 
         // Insert the code blob to the writeset.
-        for (key, blob) in to_be_published_modules.into_iter() {
-            write_set.push(((&key).into(), WriteOp::Value(blob)));
+        if write_set
+            .len()
+            .checked_add(to_be_published_modules.len())
```

Comment:
As mentioned above, it seems weird to do a checked_add and then ignore the normal result of the add.